lib/format_erf.c

```diff
@@ -736,85 +836,29 @@ static int erf_write_packet(libtrace_out_t *libtrace,
 	if (!packet->header) {
 		return -1;
 	}
-	
 
-	/* If we've had an rxerror, we have no payload to write - fix
-	 * rlen to be the correct length 
-	 */
-	/* I Think this is bogus, we should somehow figure out
-	 * a way to write out the payload even if it is gibberish -- Perry */
-	if (payload == NULL) {
-	        unsigned int pad = 0;
-	        pad = erf_get_padding(packet);
-		dag_hdr->rlen = htons(dag_record_size + pad);
-		
-	} 
-	
 	if (packet->type == TRACE_RT_DATA_ERF) {
-			numbytes = erf_dump_packet(libtrace,
-				(dag_record_t *)packet->header,
-				trace_get_framing_length(packet),
-				payload,
-                                trace_get_capture_length(packet)
-				);
+		// calculate padding to make sure eveything is 8 byte aligned
+		erf_calculate_padding(packet, &framinglen, &caplen, &padding);
+		dag_hdr->rlen = htons(framinglen + caplen + padding);
+		return erf_dump_packet(libtrace,
+				       dag_hdr,
+				       framinglen,
+				       payload,
+				       caplen,
+				       padding);
 	} else {
 		dag_record_t erfhdr;
-		int rlen;
-                int framing;
-		/* convert format - build up a new erf header */
-		/* Timestamp */
-		erfhdr.ts = bswap_host_to_le64(trace_get_erf_timestamp(packet));
-
-		/* Flags. Can't do this */
-		memset(&erfhdr.flags,1,sizeof(erfhdr.flags));
```

Comment:
The fact that we were memset()ing to 1 here in the file is concerning?
We need to check the compatibility with the erf file format going forwards and backwards between dag tools and the libtrace code before this change.